vipps-psp-api.md

```diff
@@ -204,6 +204,123 @@ Note 93 is for when the makePayment request from Vipps contains the statuses CAN
+  }
+}
+```
+
+# Recurring PSP payments
+
+The PSP API supports recurring payments out of the box. This enables the PSP to perform recurring payments through Vipps, while retaining full transactional control. This has been built as an extension to the existing PSPv2 API, and no existing integrations will be affected, other than the possibility to initialize and preform recurring payments.
+
+## Subscription and oneclick scopes
+
+As of now, there are two possible ways to perform a recurring payment - *subscription* and *oneclick*. These are referred to as the *scope* of the recurring agreement. Only one *scope* can be used at a time, and it's not possible to change the scope of an agreement.
+
+*Subscription* based payments are created as a consent to an agreement that allows the PSP to withdraw money on unknown intervals. This implies that the user won't have to accept the payment on each occasion, only the first one when consenting to the agreement. An example of this could be a subscription to a music streaming service.
+
+*Oneclick* payments are created as agreements where the user will be prompted on each payment. An example of a oneclick payment might be a ticketing service where the user will get prompted to buy a new ticket when the old one expires. Instead of switching to the app to accept the payment, the user already has a oneclick recurring agreement with Vipps as the payment source, and the ticketing app won´t need to involve the Vipps app at all.
+
+## Initialize a recurring payment
+
+Initializing a recurring payment works in the same way as a non-recurring payment, but with the inclusion of a *scope* and *agreementURL* in the init call.
+
+The *scope* can at the time be set to either *subscription* or *oneclick*. The *agreementURL* should be a link to where the user can click to administer the agreement.
+
+To start the initialization, create a standard /init call with the addition of the required fields. If you want to initialize a recurring *subscription*, the init request body could look like this.
+
+```json
+{
+  "pspRedirectUrl": "yourRedirectUrl",
+  "amount": 1337,
+  "makePaymentToken": "tokenGoesHere",
+  "makePaymentUrl": "yourCallbackEndpointUrl",
+  "currency": "NOK",
+  "merchantOrderId": "123123123",
+  "isApp": false,
+  "pspTransactionId": "{{psptransactionid}}",
+  "paymentText": "Order id: 213213",
+  "scope": "subscription",
+  "agreementURL": "linkToTheAgreement"
+}
+```
+
+In the same way as a normal, non-recurring PSPv2 payment, the PSP will receive a *makePayment* callback. In the body of this callback, you will now also find a *userToken*.
+
+## The userToken
+
+The user token is a token generated when the user has given a consent. This token is provided to the PSP in the makePayment callback when initializing or preforming a recurring payment.
+
+A *userToken* contains the *scope* of the consent, in the claim named *scope*. The token also includes various useful information
```

Comment:
Is it still called "scope"? I thought I heard that we are changing that…

Reply:
Yes still called scope.